### DISCUSSION TITLE
starship: command not found

### DISCUSSION THREAD

**Discussion by wiraki** [starship/starship#3231] [Q&A] [answered]
Ok, I feel stupid, I don't know how can that be a problem but I am running into it. 

This is on a system where I do not have root access, therefore I have install starship with the following:
`sh -c "$(curl -fsSL https://starship.rs/install.sh)" -- --bin-dir ~/bin`

And I have added the relevant part at the bottom of my .bashrc
```
> tail -3 ~/.bashrc
# Start Starship prompt
eval "$(starship init bash)"
```

I imagine the problem is that my `.bash_profile` sources '.bashrc' before exporting `$PATH` ?
```
# .bash_profile

# Get the aliases and functions
if [ -f ~/.bashrc ]; then
	. ~/.bashrc
fi

# User specific environment and startup programs

PATH=$PATH:$HOME/bin

export PATH
```

Am I supposed to make sure the path is modified before bashrc is read? I always thought the best practice is to modify path in .bash_profile at the bottom?

Thanks in advance for the help!

**Comment by davidkna** [comment, marked as answer]
I'm not aware of any best practices concerning at what point `PATH` is exported, but `~/bin` will need to be in added to your path `PATH` before you can execute the starship-binary with its name in bash.
Alternatively, you can specify the binary path directly and bypass `PATH` with `eval "$(~/bin/starship init bash)"`.
